lib/account/send_message_helper.dart

```diff
@@ -1012,11 +1029,46 @@ class SendMessageHelper {
     );
     final encoded = await jsonBase64EncodeWithIsolate(attachmentMessage);
     await _messageDao.updateAttachmentMessageContentAndStatus(
-        messageId, encoded);
+      messageId,
+      encoded,
+      attachmentResult.keys,
+      attachmentResult.digest,
+    );
     await _jobDao.insertSendingJob(messageId, conversationId);
   }
 
-  Future<AttachmentResult?> _checkAttachment(String content) async {
+  Future<AttachmentResult?> _checkAttachmentExtra(Message message) async {
+    AttachmentExtra? attachmentExtra;
+    try {
+      attachmentExtra = AttachmentExtra.fromJson(
+          await jsonDecodeWithIsolate(message.content ?? '')
+              as Map<String, dynamic>);
+    } catch (e) {
+      w('check attachment extra error: $e, content: ${message.content}');
+      attachmentExtra = null;
+    }
+    if (attachmentExtra == null) {
+      return null;
+    }
+    final createdAt = attachmentExtra.createdAt;
+    if (createdAt != null) {
+      final date = DateTime.tryParse(createdAt);
```

Comment:
it's better use `DateTime.tryParse(createdAt)?.toLocal()` ? since `DateTime.now()` is a local date.